helmfile/upstream/prometheus-community/kube-prometheus-stack/charts/kube-state-metrics/values.yaml

```diff
@@ -233,7 +233,8 @@ metricDenylist: []
 # label keys you would like to allow for them (Example: '=namespaces=[k8s-label-1,k8s-label-n,...],pods=[app],...)'.
 # A single '*' can be provided per resource instead to allow any labels, but that has
 # severe performance implications (Example: '=pods=[*]').
-metricLabelsAllowlist: []
+metricLabelsAllowlist:
+  - nodes=[topology.kubernetes.io/zone]
```

Comment:
Isn't it possible to set this in the kube-prometheus-stack values file, e.g. [here](https://github.com/elastisys/compliantkubernetes-apps/blob/main/helmfile/values/kube-prometheus-stack-wc.yaml.gotmpl#L23)?

Reply:
Good question. Still new to all these layers of indirection 😄 

Something along these lines perhaps?

``` diff
@@ -21,6 +21,13 @@ prometheusOperator:
     resources: {{- toYaml .Values.prometheusOperator.prometheusConfigReloader.resources | nindent 6 }}
 
 kube-state-metrics:
+
+  # if zones enabled
+  {{- if true }}
+  metricLabelsAllowlist:
+  - nodes=[topology.kubernetes.io/zone]
+  {{- end }}
+
   resources: {{- toYaml .Values.kubeStateMetrics.resources | nindent 4 }}
   {{- if .Values.kubeStateMetrics.clusterAPIMetrics.enabled }}
   rbac:
```

Is there an existing settings for whether zones are used at all yet?

Reply:
> Is there an existing settings for whether zones are used at all yet?

No, not that i'm aware of

Reply:
Hm, does it even need to be conditional?

Reply:
Not entirely sure, you could see what happens if the label isn't set on the nodes.

Reply:
If it's in the allow-list but no such labels are set it just doesn't show up in the metrics data, so I don't think there'll be any storage overhead for clusters without it. Unless there's some significant optimization for the case of an _empty_ allow-list, I doubt there's much of a performance impact either.

So not adding conditional or new settings for now then.